data/crac/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/InjectionSetpointAdderImplTest.java

```diff
@@ -54,14 +56,28 @@ void testOk() {
     @Test
     void testNoNetworkElement() {
         InjectionSetpointAdder injectionSetpointAdder = networkActionAdder.newInjectionSetPoint()
-            .withSetpoint(100.);
+            .withSetpoint(100.).withUnit(Unit.MEGAWATT);
         assertThrows(FaraoException.class, injectionSetpointAdder::add);
     }
 
     @Test
     void testNoSetpoint() {
         InjectionSetpointAdder injectionSetpointAdder = networkActionAdder.newInjectionSetPoint()
-            .withNetworkElement("groupNetworkElementId");
+            .withNetworkElement("groupNetworkElementId").withUnit(Unit.MEGAWATT);
         assertThrows(FaraoException.class, injectionSetpointAdder::add);
     }
+
+    @Test
+    void testNoUnit() {
+        InjectionSetpointAdder injectionSetpointAdder = networkActionAdder.newInjectionSetPoint()
+                .withNetworkElement("groupNetworkElementId").withSetpoint(100.);
+        assertThrows(FaraoException.class, injectionSetpointAdder::add);
```

Comment:
check the message in the exception, so we're sure this call is failing for the right reason. We did not do this before, but it's a good practive we're trying to follow from now on:
```suggestion
        Exception e = assertThrows(FaraoException.class, injectionSetpointAdder::add);
        assertEquals("...", e.getMessage());
```

Reply:
Done !